src/shared/use-api-attribute-params.js

```diff
@@ -11,30 +11,34 @@ export const useApiAttributeParams = () => {
     const { data } = useMetadata()
     const [{ dataSetId, attributeOptionComboSelection }] = useContextSelection()
 
-    if (data && dataSetId) {
-        const dataSet = selectors.getDataSetById(data, dataSetId)
-        if (dataSet === null || dataSet === undefined) {
-            return
-        }
-        const catComboId = dataSet.categoryCombo.id
-        const categoryCombo = selectors.getCategoryComboById(data, catComboId)
+    return useMemo(() => {
+        if (data && dataSetId) {
+            const dataSet = selectors.getDataSetById(data, dataSetId)
+            if (dataSet === null || dataSet === undefined) {
+                return
+            }
+            const catComboId = dataSet.categoryCombo.id
+            const categoryCombo = selectors.getCategoryComboById(
+                data,
+                catComboId
+            )
 
-        // Sort these to produce consistent query and mutation keys
-        const selectedOptions = Object.values(
-            attributeOptionComboSelection
-        ).sort()
+            // Sort these to produce consistent query and mutation keys
+            const selectedOptions = Object.values(
+                attributeOptionComboSelection
+            ).sort()
 
-        if (!categoryCombo.isDefault) {
-            // we don't need to supply attributeCombo/options when it's default
-            return {
-                attributeCombo: catComboId,
-                attributeOptions: selectedOptions,
+            if (!categoryCombo.isDefault) {
+                // we don't need to supply attributeCombo/options when it's default
+                return {
+                    attributeCombo: catComboId,
+                    attributeOptions: selectedOptions,
+                }
             }
         }
-    }
-
-    return {
-        attributeCombo: undefined,
-        attributeOptions: undefined,
-    }
+        return {
+            attributeCombo: undefined,
+            attributeOptions: undefined,
+        }
+    }, [data, dataSetId, attributeOptionComboSelection])
```

Comment:
I think `attributeOptionComboSelection` itself won't be stable (see @Mohammer5's note here: https://github.com/dhis2/aggregate-data-entry-app/blob/554025ccca2063af53d20c77b65e27b8f9ada478/src/shared/use-context-selection/use-manage-inter-param-dependencies.js#L168)?

Reply:
From my testing this is not the case. 

```
    const { attributeOptionComboSelection } = useContextSelection()
    useEffect(() => {
        console.log('run useffect')
    }, [attributeOptionComboSelection])
```

Only runs when the option is changed.

Reply:
Cool. Well that makes things easier 🙂